lib-src/rpc2/include/rpc2/pack_helper.h

```diff
@@ -0,0 +1,106 @@
+/* BLURB lgpl
+
+                           Coda File System
+                              Release 5
+
+          Copyright (c) 1987-2016 Carnegie Mellon University
+                  Additional copyrights listed below
+
+This  code  is  distributed "AS IS" without warranty of any kind under
+the  terms of the  GNU  Library General Public Licence  Version 2,  as
+shown in the file LICENSE. The technical and financial contributors to
+Coda are listed in the file CREDITS.
+
+                        Additional copyrights
+
+#*/
+
+/*
+                         IBM COPYRIGHT NOTICE
+
+                          Copyright (C) 1986
+             International Business Machines Corporation
+                         All Rights Reserved
+
+This  file  contains  some  code identical to or derived from the 1986
+version of the Andrew File System ("AFS"), which is owned by  the  IBM
+Corporation.   This  code is provided "AS IS" and IBM does not warrant
+that it is free of infringement of  any  intellectual  rights  of  any
+third  party.    IBM  disclaims  liability of any kind for any damages
+whatsoever resulting directly or indirectly from use of this  software
+or  of  any  derivative work.  Carnegie Mellon University has obtained
+permission to  modify,  distribute and sublicense this code,  which is
+based on Version 2  of  AFS  and  does  not  contain  the features and
+enhancements that are part of  Version 3 of  AFS.  Version 3 of AFS is
+commercially   available   and  supported  by   Transarc  Corporation,
+Pittsburgh, PA.
+
+*/
+
+#ifndef _PACK_HELPER_
+#define _PACK_HELPER_
+#include <rpc2/rpc2.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define STUBCLIENT 0
+#define STUBSERVER 1
+#define STUBIN 1
+#define STUBOUT 2
+#define _PAD(n)((((n)-1) | 3) + 1)
+
+
+int unpack_int(RPC2_Integer* new_ptr, char** old_ptr, char* EOB);
```

Comment:
I think I would prefer the signature to be closer to the unix stdio, i.e. move the buffer and end-of-buffer to the front (similar to an fd/FILE*) and the to be packed/returned values to the end of the argument list. The ordering becomes clearer once you go down the list and there are lengths and modes involved.